typing/typetexp.ml

```diff
@@ -687,9 +687,17 @@ let transl_simple_type_univars env styp =
 
 let transl_simple_type_delayed env styp =
   univars := []; used_variables := TyVarMap.empty;
+  begin_def ();
   let typ = transl_type env Extensible styp in
+  end_def ();
   make_fixed_univars typ.ctyp_type;
-  (typ, globalize_used_variables env false)
+  (* This brings the used variables to the global level, but doesn't link them
+     to their other occurences just yet. This will be done when [force] is
+     called. *)
+  let force = globalize_used_variables env false in
+  (* Generalizes everything except the variables that were just globalized. *)
```

Comment:
This would be more readable if `force` was eta-expanded: `let force () = ...`.

Reply:
But it would be incorrect: here we are relying on some side effects happening right now, before the function is created (not that this was always the case, it is just made more obvious by this PR).

Implicit currying strikes again.

Reply:
Apologies, I tried to look for this eventuality but I messed up my own checking of the source.